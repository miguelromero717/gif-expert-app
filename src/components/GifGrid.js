import React, { useEffect, useState } from 'react'
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {
        const url = "https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=ryryBQQHHBWSRuDV6QJaHUOt69jskdDi";
        const response = await fetch(url);
        const { data } = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);

        setImages(gifs);
    }

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

export default GifGrid;
