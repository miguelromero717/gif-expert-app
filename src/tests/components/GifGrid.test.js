import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid tests', () => {
    test('should render component correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const category = 'Matrix';
        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot();
    })

    test('should show items when load images using useFetchGifs hook', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/image/any.jpg',
            title: 'Any Image Title'
        },
        {
            id: '123',
            url: 'https://localhost/image/any.jpg',
            title: 'Any Image Title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const category = 'Matrix';
        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })

})
