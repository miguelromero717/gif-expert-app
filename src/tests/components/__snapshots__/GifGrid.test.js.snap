// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should render component correctly 1`] = `
<Fragment>
  <h3>
    Matrix
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
